LXStudio-IDE/src/main/java/titanicsend/util/PanelStriper.java

```diff
@@ -0,0 +1,145 @@
+package titanicsend.util;
+
+import heronarts.lx.model.LXPoint;
+import titanicsend.model.TEVertex;
+
+import java.util.*;
+
+import static java.lang.Math.cos;
+
+public class PanelStriper {
+  public static final int MARGIN = 75000; // 75k microns ~= 3 inches
+  public static final int DISTANCE_BETWEEN_PIXELS = 50000; // 25k microns ~= 1 inch
+
+  public static List<LXPoint> stripe(TEVertex v0, TEVertex v1, TEVertex v2) {
+    FloorTransform floorTransform = new FloorTransform(v0, v1, v2);
+
+    List<FloorPoint> floorPoints = stripeFloor(
+            floorTransform.f0, floorTransform.f1, floorTransform.f2);
+
+    List<LXPoint> rv = new ArrayList<LXPoint>();
+    for (FloorPoint f : floorPoints) {
+      rv.add(floorTransform.fly(f));
+    }
+
+    return rv;
+  }
+
+  private static List<FloorPoint> stripeFloor(FloorPoint f0, FloorPoint f1, FloorPoint f2) {
+    FloorPoint currentPoint = findStartingPoint(f0, f1, f2);
+    ArrayList<FloorPoint> rv = new ArrayList<FloorPoint>();
+
+    // Calc the angle we set off at to get from near f0 -> f1
+    double heading = calcHeading(f0, f1);
+
+    // When we get to f1 (or, at least, hit the margin),
+    // we bump over in this direction (toward f2) before
+    // spinning 180 degrees and making a new stripe.
+    double endOfRowHeading = calcHeading(f1, f2);
+
+    // And then when we reach the end of the second row,
+    // we again bump towards f2, but now the angle will
+    // be different because we're in the neighborhood of f0.
+    double endOfRowHeadingNext = calcHeading(f0, f2);
+
+    final int MAX_POINTS = 2500;  // Sanity check
+
+    for (int i = 0; i < MAX_POINTS; i++) {
+      rv.add(currentPoint);
+
+      double nextX = currentPoint.x + DISTANCE_BETWEEN_PIXELS * Math.cos(heading);
+      double nextZ = currentPoint.z + DISTANCE_BETWEEN_PIXELS * Math.sin(heading);
+      FloorPoint nextPoint = new FloorPoint(nextX, nextZ);
+      if (distanceToEdge(f0, f1, f2, nextPoint) >= MARGIN) {
+        // We haven't yet reached the margin.
+        currentPoint = nextPoint;
+      } else {
```

Comment:
just to check my understanding, we're not accomodating for the leds between rows here, right. Does that matter? I'm imagining probably not unless we have a very specific row-by row pattern for a panel..

Reply:
Not yet; I wanted to confirm the conditions in the field first so we don't have to code it up twice.